mtbf_driver/mtbf.py

```diff
@@ -205,16 +206,29 @@ def deinit(self):
         if os.path.exists(info):
             shutil.copy2(info, self.workspace)
 
+        archive_file = os.path.basename(self.archive_folder) + ".zip"
+        try:
+            os.remove(archive_file)
+        except OSError:
+            pass
+        with ZipFile(archive_file, "w") as archive:
+            for root, dirs, files in os.walk(self.archive_folder):
+                for f in files:
+                    f = os.path.join(root, f)
+                    archive.write(f, os.path.relpath(f, self.archive_folder))
+
```

Comment:
with statement implicitly introduce close when exiting the block
ref: https://docs.python.org/2/library/zipfile.html
"ZipFile is also a context manager and therefore supports the with statement. In the example, myzip is closed after the with statement’s suite is finished—even if an exception occurs:"